src/docs/sdk/features.mdx

```diff
@@ -231,6 +231,58 @@ If Performance Monitoring is both supported by the SDK and enabled in the client
 - span status must match HTTP response status code ([see Span status to HTTP status code mapping](/sdk/event-payloads/span/))
 - when network error occurs, span status must be set to `internal_error`
 
+### HTTP Client errors
+
+HTTP Client errors are automatically captured and sent to `https://sentry.io`.
+
+The HTTP Client integration should have 3 configuration options:
+
+* `captureFailedRequests` defaults to `false` due to PII reasons.
+* `failedRequestStatusCodes` defaults to `500` to `599`, this configuration option accepts a `List` of `HttpStatusCodeRange` which is a range of HTTP status code -> `min` to `max` or a single `status_code`.
+* `failedRequestTargets` defaults to `.*`, this configuration option accepts a `List` of `String` that may be Regular expressions as well, similar to <Link to="/sdk/performance/#tracepropagationtargets">tracePropagationTargets</Link>.
+
+Then the HTTP Client integration should not capture the error:
+
+* If `captureFailedRequests` is disabled.
+* If the HTTP Request url isn't a match for any of the `failedRequestsTargets`.
+* If the HTTP Response status code isn't within the defined ranges in `failedRequestStatusCodes`.
+
+The HTTP Client integration should capture error events with the following properties:
+
+The Request interface, See the <Link to="/sdk/event-payloads/request/">Spec</Link> for details.
+
+The Response context (Not yet under the Spec):
+
+```json
+{
+  "contexts": {
+    "response": {
+      "type": "response",
+      "cookies": "PHPSESSID=298zf09hf012fh2; csrftoken=u32t4o3tb3gg43; _gat=1;",
+      "headers": {
+        "content-type": "text/html"
+        /// ...
+      },
+      "status_code": 500,
+      "body_size": 1000, // in bytes
+    }
+  }
+}
+```
+
+The Exception Interface, See the <Link to="/sdk/event-payloads/exception/">Spec</Link> for details.
+
+If the HTTP Client integration does not throw an error for unsuccessful requests, you can create a synthetic exception following this Spec:
+
+* Set the <Link to="/sdk/event-payloads/exception/#exception-mechanism">Exception Mechanism</Link> with a proper `type` such as `SentryOkHttpInterceptor`.
+* Set the <Link to="/sdk/event-payloads/stacktrace/">Stack Trace Interface</Link> with `snapshot=true`.
+
+When capturing error events, pass the original `Request` and `Response` objects from the HTTP Client as `hints`, so the users may filter out events with the full context.
```

Comment:
`l`: I would mention `beforeSend`. What SDKs not having hints? Is it fine to not do this?
```suggestion
When capturing error events, pass the original `Request` and `Response` objects from the HTTP Client as `hints`, so the users may filter out events in `beforeSend` with the full context.
```

Reply:
I still would mention `beforeSend`.